src/Payum/Paypal/ExpressCheckout/Nvp/Action/Api/CancelRecurringPaymentsProfileAction.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace Payum\Paypal\ExpressCheckout\Nvp\Action\Api;
+
+use Payum\Core\Request\Cancel;
+use Payum\Core\Bridge\Spl\ArrayObject;
+use Payum\Core\Exception\RequestNotSupportedException;
+use Payum\Paypal\ExpressCheckout\Nvp\Request\Api\CancelRecurringPaymentProfile;
+
+class CancelRecurringPaymentsProfileAction extends BaseApiAwareAction
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function execute($request)
+    {
+        /** @var $request Cancel */
+        RequestNotSupportedException::assertSupports($this, $request);
+
+        $model = ArrayObject::ensureArrayObject($request->getModel());
+        $model->validateNotEmpty(array('PROFILEID', 'ACTION'));
+
+        $model['ACTION'] = Api::RECURRINGPAYMENTACTION_CANCEL;
+
+        $model->replace(
+            $this->api->manageRecurringPaymentsProfile((array) $model)
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supports($request)
+    {
+        if (false == ($request instanceof Cancel && $model = $request->getModel() instanceof \ArrayAccess)) {
+            return false;
+        }
+
+        // it must take into account only recurring payments, common payments must be cancelled by another action.
+        return isset($model['BILLINGPERIOD']);
```

Comment:
nope, because there maybe a cancel request for common payment (not recurring one) and in that case second action would name work.  It must be in supports method. You can use any other fields that allows you different common payment from recurring one.